### ISSUE TITLE
Incorrect transformation of math

### ISSUE THREAD

**Issue by jxxcarlson** [asciidoctor/asciidoctor#1058]
Consider the file `foo.ad` below.  When I apply asciidoctor (1.5, rc5) to it, the
relevant part of the generated HTML  file is as in `foo.html`.  In part (2),
the HTML is incorrect --- rather than being passed through, the text
delimited by `\[  ... \]` is processed. The two tags `<sup>` should not
be there. (It makes a significant difference in the rendered text.)

```
foo.ad
------
:stem: latexmath

== Test

(1)
\[ V = s^3  =  10^{63} \text{m}^3 \]

(2)
\[ V = s^3  =  (10^{21})^3 \text{m}^3 \]
------
```

``` foo.html
------
<div class="paragraph">
<p>(1)
\[ V = s^3  =  10^{63} \text{m}^3 \]</p>
</div>
<div class="paragraph">
<p>(2)
\[ V = s^3  =  (10<sup>{21})</sup>3 \text{m}^3 \]</p>
</div>
------
```

**Comment by andreabedini** [comment]
Hi there, I am writing here because I think it's related.

The [Asciidoctor manual](http://asciidoctor.org/docs/user-manual/#using-multiple-stem-interpreters) and compatibility with [asciidoc](http://asciidoc.org/userguide.html#X78) suggest that `stem:[V = s^3  =  10^{63} \text{m}^3]` should be rather written `stem:[$V = s^3  =  10^{63} \text{m}^3$]`, note the extra dollars. But with Asciidoctor 1.5.2 if I do `echo 'latexmath:[$V = s^3  =  10^{63} \\text{m}^3$]' | asciidoctor -a stem -`

that gives

```
<p>\($V = s^3  =  10^{63} \text{m}^3$\)</p>
```
note the extra dollars. Bug?

**Comment by mojavelinux** [comment]
The manual is incorrect here. In Asciidoctor, you never need to include the math delimiters if you are using the `stem` macros (including `asciimath` and `latexmath`) because Asciidoctor is smart enough to add the correct delimiters. The way it should be written is simply:

```
stem:[V = s^3 = 10^{63} \text{m}^3]
```

or

```
latexmath:[V = s^3 = 10^{63} \text{m}^3]
```

I'll fix the user manual. Thanks for pointing this out!!

**Comment by fego** [comment]
Hello,

I have a related issue I think. I just tried the documentation examples and the formulas are not transformed by MathJax. See https://gist.github.com/4fc565a5f8223aeb6926.git

**Comment by mojavelinux** [comment]
That gist is not available. Can you verify the link?

**Comment by fego** [comment]
Sorry, here is the correct URL : https://gist.github.com/4fc565a5f8223aeb6926.git 
Thks

**Comment by abelsromero** [comment]
Still can't see it, maybe the gist is not public?

**Comment by mojavelinux** [comment]
I've had this problem before. It seems to be a bug in Gist that it copies URLs wrong.

Here's the publicly-accessible URL:

https://gist.github.com/fego/4fc565a5f8223aeb6926

**Comment by mojavelinux** [comment]
Notice that I added the username to the URL. It seems when you are logged in, the URL to your own gists drops the part that allows other people to see it. Shame on gist.

**Comment by mojavelinux** [comment]
...and the .git extension as well. Either way, we've got it now!

**Comment by mojavelinux** [comment]
@fego The MathJax configuration is being output incorrectly in your environment, which in turn is breaking MathJax.

Here's what you are getting:

```
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [\(\)],
    displayMath: [\[\]],
    ignoreClass: "nostem|nolatexmath"
  },
  asciimath2jax: {
    delimiters: [\$\$],
    ignoreClass: "nostem|noasciimath"
  },
  TeX: { equationNumbers: { autoNumber: "none" } }
});
```

Here's what it should be:

```
MathJax.Hub.Config({
  tex2jax: {
    inlineMath: [["\\(", "\\)"]],
    displayMath: [["\\[", "\\]"]],
    ignoreClass: "nostem|nolatexmath"
  },
  asciimath2jax: {
    delimiters: [["\\$", "\\$"]],
    ignoreClass: "nostem|noasciimath"
  },
  TeX: { equationNumbers: { autoNumber: "none" } }
});
```

And here is probably the reason:

```
ruby 1.8.7
```

This was a know issue with Ruby 1.8.7 which may be resolved in 1.5.3. However, a better solution is to just upgrade Ruby because Ruby 1.8.7 is EOL and we're dropping support for it in Asciidoctor 1.6.0 (and maybe sooner).

**Comment by mojavelinux** [comment]
Yep, I just confirmed that this was fixed by the following commit, pending for 1.5.3.

https://github.com/asciidoctor/asciidoctor/commit/f86e65d92bf0af5e05d30d1983b57daa57286b51

**Comment by fego** [comment]
Great, thank you for your answer and patience :)
